challenge_2/c/xoumi/SingleDigit.c

```diff
@@ -0,0 +1,24 @@
+#include<stdio.h>
+#include<string.h>
+void main()
+{
+    char array[255],i;
+    int flag,len,j;
+    printf("Enter array of numbers: ");
+    gets(array);
+    for(len=0;array[len]!='\0';len++);
+    len-=1;
+
+    //Run every number from 0 to 9 and if matching, flag+=1
+    //This has a runtime of O(11n) I think. That's linear right ?
```

Comment:
*facepalms
Well you're right, neither of these work for anything more than two digit numbers.
This entire algorithm fails if I cross single digit numbers...
back to drawing board I guess. Thanks.